Quaver.Shared/Screens/Gameplay/Rulesets/Keys/GameplayRulesetKeys.cs

```diff
@@ -61,6 +69,20 @@ public bool ShowHits
 
                 _showHits = value;
 
+                var manager = (HitObjectManagerKeys) HitObjectManager;
+                foreach (GameplayHitObjectKeys hitObject in manager.ActiveNoteLanes.Concat(manager.DeadNoteLanes).Concat(manager.HeldLongNoteLanes).Flatten())
+                {
+                    var tint = hitObject.Tint * (_showHits ? 1 : ShowHitsNoteAlpha);
+                    var newTint = hitObject.Tint * (_showHits ? ShowHitsNoteAlpha : 1);
+
+                    hitObject.HitObjectSprite.Tint = tint;
+                    hitObject.HitObjectSprite.FadeToColor(newTint, Easing.OutQuad, 250);
```

Comment:
You'll want to call `hitObject.[part_of_object].ClearAnimations()` before adding a new one, so you only have one fade animation running at a time.